doc/analysis.md

```diff
@@ -0,0 +1,55 @@
+# Analyzing Bitcoin Core
+
+## Static analysis
+
+### Analyzing Bitcoin Core using Clang Static Analyzer
+
+#### Quickstart guide
+
+```
+$ git clone https://github.com/bitcoin/bitcoin
+$ cd bitcoin/
+$ ./autogen.sh
+$ CC=clang CXX=clang++ ./configure --with-incompatible-bdb --disable-ccache
+$ scan-build --use-cc=clang --use-c++=clang++ make
```

Comment:
If this is OS-specific, I think it's necessary to state that. It seems to be, from looking at https://clang-analyzer.llvm.org/ and https://clang-analyzer.llvm.org/scan-build.html after seeing `scan-build: 'ccc-analyzer' does not exist at '/usr/local/bin/ccc-analyzer'` here and not finding a linux version.